src/server/subscriptions.ts

```diff
@@ -0,0 +1,49 @@
+import * as uuidv4 from "uuid/v4";
+import { EventEmitter } from "events";
+import { augurEmitter } from "../events";
+import { ErrorCallback } from "../types";
+
+export class Subscriptions extends EventEmitter {
+  public subscribe(eventName: string, params: any, publish: (data: {}) => void): string {
+    switch (eventName) {
+      case "MarketCreated":
+        return this.subscribeToEvent(eventName, params, publish);
+      default:
+        throw new Error(`Event ${eventName} not available for subscription`);
+    }
+  }
+
+  public unsubscribe(subscription: string): void  {
+    this.emit(`unsubscribe:${subscription}`);
+  }
+
+  public removeAllListeners(eventName?: string | symbol | undefined): this {
+    this.emit("removeAllListeners");
+    return super.removeAllListeners(eventName);
+  }
+
+  private subscribeToEvent(eventName: string, params: any, publish: (data: {}) => void): string {
+    const subscription: string = uuidv4();
+
+    const handler = (data: {}): void => {
+      this.emit(eventName, data);
+    };
```

Comment:
I think you need to either `.bind(this)` or (maybe cleaner?) inline the `this.emit` call.  i.e.,
```
const handler = (data: {}): void => { this.emit(eventName, data); }.bind(this);
```
or
```
const handler = (data: {}): void => this.emit(eventName, data);
```

Reply:
Fat arrow syntax in TS/ES6 will automatically bind `this` to the the outer scope. But yes to putting it on one line!

Reply:
leaving the braces so we throw away the return value

Reply:
Right, I use `this` so rarely that I wasn't even aware of this fat-arrow function feature...